### ISSUE TITLE
Simplify binding when there are a lot of classes/methods/functions to bind

### ISSUE THREAD

**Issue by qtnc** [blueloveTH/pocketpy#184]
Hello,

I was using AngelScript for a project, and might be interested to switch to PocketPy, as a scripting language for a small game.
There are several advantages in PocketPy, for example no need to implement strings and lists/arrays yourself, and the fact that it's (almost) the well known python instead of a specific language.

One of the things AngelScript can do that PockerPy can't do currently, is to bind a function just in one line. There's no need to write a lambda wrapper for each of the functions/methods you want to make available in scripts.
This makes much less work when you have to register many classes/method/functions.

The idea would be to go towards something like this:

    struct Point {
      double x;
      double y;
      
      Point (double x1, double y1): x(x1), y(y1) {}
      
      double length () {
        return sqrt(x*x + y*y);
      }
      
      void add (const Point& pt) {
        x += pt.x;
        y += pt.y;
      }
      
      PY_CLASS(Point, builtins, Point, vm, mod, type){
        vm->bind_constructor<Point, double, double>(type, "__new__(self, x=0, y=0)");
        vm->bind_field(type, "x", &Point::x);
        vm->bind_field(type, "y", &Point::y);
        vm->bind_method(type, "length(self)", &Point::length);
        vm->bind_method(type, "add(self, other)", &Point::add);
      }
      
    };

Basically it would use some template obscure magic to translate from:

    vm->bind_method(type, "length(self)", &Point::length);

to approximatively:

    vm->bind(type, "length(self)", [](VM* vm, ArgsView args){
      Point& self = py_cast<Point&>(vm, args[0]);
      return py_var( self.length() );
    });

I may share some code if interested.

**Comment by blueloveTH** [comment]
Hi @qtnc
First, please allow me to invite you to join our discord server https://discord.gg/WWaq72GzXv I am glad to provide suggessions for your bindings of your game.

Currently, the officially supported bindings are like this:
```cpp
vm->bind(type, "length(self)", [](VM* vm, ArgsView args){
  Point& self = py_cast<Point&>(vm, args[0]);
  return py_var( self.length() );
});
```

We want to **make users know what happens inside the binding**. And this is easy to debug than templates.

I know you want to have an easier way. There are solutions.

You can check this https://pocketpy.dev/bindings/#automatic-bindings we have a repo to demonstrate how to generate bindings from metadata. And in this repo, I use some undocumented binding apis. For exmaple, `_bind(...)`.

https://github.com/blueloveTH/pkpy-bindings/blob/main/output/raylibw.cpp#L2012

```cpp
    _bind(vm, mod, "InitWindow(width: int, height: int, title: str) -> None", &InitWindow);
    _bind(vm, mod, "CloseWindow() -> None", &CloseWindow);
    _bind(vm, mod, "WindowShouldClose() -> bool", &WindowShouldClose);
    _bind(vm, mod, "IsWindowReady() -> bool", &IsWindowReady);
    _bind(vm, mod, "IsWindowFullscreen() -> bool", &IsWindowFullscreen);
    _bind(vm, mod, "IsWindowHidden() -> bool", &IsWindowHidden);
    _bind(vm, mod, "IsWindowMinimized() -> bool", &IsWindowMinimized);
    _bind(vm, mod, "IsWindowMaximized() -> bool", &IsWindowMaximized);
```

You can find many useful practice in it.

**Comment by blueloveTH** [comment]
Have a look at https://github.com/blueloveTH/pocketpy/blob/main/include/pocketpy/bindings.h for all convenient macros and templates.
